addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2218,6 +2218,9 @@ const ListUserValueWidget = UserValueWidget.extend({
         }
         this._addItemToTable(undefined, this.el.dataset.defaultValue, recordData);
         this._notifyCurrentState();
+        const scrollableEl = this.$el[0].querySelector('.o_we_table_wrapper');
+        // Scroll to the new list element.
+        scrollableEl.scrollTop = scrollableEl.scrollHeight;
```

Comment:
`this.$el[0].querySelector('tr:last-child').scrollIntoView({behavior: "smooth"});`
You could avoid the not necessary `scrollHeight` with `scrollIntoView` here. And moreover, with that, the scrolling of the snippet panel is also done automatically if needed. It also adds a "smooth" effect better for UX (the mouse arrow always stays above the button).
Video => [link](https://imgur.com/oCnUgyH)